src/Sys/InspectSeoHandler.php

```diff
@@ -0,0 +1,176 @@
+        }
+
+        if (empty($retryCommand->get('search_engines'))) {
+            return;
+        }
+
+        $retries = $command->get('ctx_retries');
+        $maxRetries = $this->flags->getInt('inspect_seo_max_retries', 5);
+
+        if ($retries < $maxRetries) {
+            $retryCommand->set('ctx_retries', $retries + 1);
+
+            $pbjx->sendAt(
+                $retryCommand,
+                strtotime($this->flags->getString('inspect_seo_retry_delay', '+15 minutes')),
+                "$nodeRef.inspect-seo"
+            );
+        }
+    }
+
+    public function checkIndexStatusForGoogle(Message $command, Pbjx $pbjx, Message $node): Message
+    {
+        if ($this->flags->getBoolean('inspect_seo_handler_google_disabled')) {
+            return $command;
+        }
+
+        $request = new \Google_Service_SearchConsole_InspectUrlIndexRequest();
+        $siteUrl = $this->flags->getString('inspect_seo_google_site_url');
+
+        if (empty($siteUrl)) {
+            $this->logger->error('Must provide a site url.', [
+                'site_url' => $siteUrl,
+            ]);
+
+            return $command;
+        }
+
+        $request->setSiteUrl($siteUrl);
+        $url = UriTemplateService::expand(
+            "{$node::schema()->getQName()}.canonical", $node->getUriTemplateVars()
+        );
+        $request->setInspectionUrl($url);
+        $client = new \Google_Client();
+        $client->addScope(\Google_Service_SearchConsole::WEBMASTERS_READONLY);
+
+        $response = null;
+
+        try {
+            $base64EncodedAuthConfig = Crypto::decrypt(getenv('GOOGLE_SEARCH_CONSOLE_API_SERVICE_ACCOUNT_OAUTH_CONFIG'), $this->key);
+            $client->setAuthConfig(json_decode(base64_decode($base64EncodedAuthConfig), true));
+            $service = new \Google_Service_SearchConsole($client);
+            $response = $service->urlInspection_index->inspect($request);
+        } catch (\Throwable $e) {
+            $this->logger->error('An error occurred in checkIndexStatus for google.', [
+                'exception' => $e,
+                'node_ref' => $node->generateNodeRef(),
+                'url' => $url,
+                'retry_count' => $command->get('ctx_retries'),
+                'stack_trace' => $e->getTraceAsString(),
+            ]);
+        }
+
+        if ($response === null) {
+            return $command;
+        }
+
+        $inspectionResult = $response->getInspectionResult();
+        $indexStatusResult = $inspectionResult->getIndexStatusResult() ?? null;
+        $ampResult = $inspectionResult->getAmpResult() ?? null;
+
+        $isConclusiveForWeb = $indexStatusResult && ($indexStatusResult->getVerdict() === 'PASS' || $indexStatusResult->getVerdict() === 'FAIL');
+        $isConclusiveForAmp = !$node::schema()->hasField('amp_enabled') || ($ampResult && ($ampResult->getVerdict() === 'PASS' || $ampResult->getVerdict() === 'FAIL'));
```

Comment:
@efjacobson @digitalartcrew didn't google cancel amp, like literally it's phased out or in process?

Reply:
amp is supported by the api and no news on the site page: https://developers.google.com/amp

Reply:
@gdbrown i think the world at large is phasing it out, yes, but maybe no one told google (or an adops team that i happen to be acquainted with) yet

Reply:
@efjacobson that thing really needs to die.  we killed it on our end, if your mobile is fast enough it won't affect your presence in google carousels and serp what not.  however, we can leave it running i guess.  people love it https://www.reddit.com/r/SEO/comments/15vz9bj/is_amp_really_dead/